app/src/main/java/at/bitfire/icsdroid/UriUtils.kt

```diff
@@ -47,7 +47,9 @@ object UriUtils {
      * @throws IllegalArgumentException if no URL is found in the string
      */
     fun String.stripUrl(): String? {
-        return "([a-zA-Z]+)://(\\w+)(.\\w+)*[\\w.&?=*]*".toRegex()
+        //      schema             host          port         path           query
+        return "([a-zA-Z]+://)" + "([\\w.]+)" + "(:\\d+)?" + "([\\w/]+)?" + "(\\?[\\w.&=*]*)?"
```

Comment:
This way you are doing `String + String + String + String + Regex`, that's why Android Studio's linter says that the function never returns null (`String?`). You have to wrap the string concatenation in paranthesis:
```kotlin
fun String.stripUrl(): String? {
    //      schema             host          port         path           query
    return ("([a-zA-Z]+://)" + "([\\w.]+)" + "(:\\d+)?" + "([\\w/]+)?" + "(\\?[\\w.&=*]*)?")
        .toRegex()
        .find(this)
        ?.value
}
```